### PR TITLE
Add precise error conditions to verifier unit tests

### LINKED ISSUE #118: Unit Test in go-cose repository needs to check for precise error condition
Current Unit test (example : file verifier_test.go) 
TestNewVerifier ( ) 

Only checks whether the negative tests have returned an Error or not ?
Then the test code compares it with a Boolean "WantErr" to understand that whether error was expected in the given test instance or not?

However the test code does not check the precise "error string" which maps to precise point in the code where the 
tests should return an error.

With this type of testing could lead to inaccurate results as code may return an error from incorrect part but the test may still indicate pass.

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**
Fixes #118

Signed-off-by: Yogesh Deshpande <yogesh.deshpande@arm.com>

### CODE REVIEW

File: verifier_test.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ import (
 	"testing"
 )
 
+const noError = ""
```
Comment (shizhMSFT) [inline verifier_test.go:15]: This variable name is shared across all files and can easily cause conflicts. It is better to just use `""` at line 129.

Comment (yogeshbdeshpande) [reply verifier_test.go]: @shizhMSFT 👍 Done!

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: verifier_test.go [hunk 2]
```diff
@@ -95,33 +95,36 @@ func TestNewVerifier(t *testing.T) {
...
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := NewVerifier(tt.alg, tt.key)
-			if (err != nil) != tt.wantErr {
+			if err != nil && (err.Error() != tt.wantErr) {
+				t.Errorf("NewVerifier() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			} else if err == nil && (tt.wantErr != "") {
```
Comment (shizhMSFT) [inline verifier_test.go:127]: nit:
```suggestion
			if (err != nil && err.Error() != tt.wantErr) || (err == nil && tt.wantErr != "") {
```

Comment (yogeshbdeshpande) [reply verifier_test.go:127]: @shizhMSFT: I had a similar thought , but then felt that the existing construct is more readable than what is been suggested in your comment, hence I preferred this style!

Comment (thomas-fossati) [reply verifier_test.go:127]: I agree with Yogesh, this check is already a bit too convoluted, we need to keep it readable.  Maybe one day we can move to stretchr/testify.

**Pull request merged**